File: shopping_basket/src/shopping_basket/in_memory_shopping_basket_repository.py
from shopping_basket.src.shopping_basket.basketitem import BasketItem
from shopping_basket.src.shopping_basket.shopping_basket_repository import ShoppingBasketRepository
from shopping_basket.src.shopping_basket.user import UserID


class InMemoryShoppingBasketRepository(ShoppingBasketRepository):
    def __init__(self):
        self.__items = []

    def add_item(self, item: BasketItem):
        self.__items.append(item)

    def basket_for(self, user_id: UserID):
        return [item for item in self.__items if item.user_id == user_id]

File: shopping-basket-java/src/main/java/shopping_basket/InMemoryShoppingBasketRepository.java
package shopping_basket;

public class InMemoryShoppingBasketRepository implements ShoppingBasketRepository{
    @Override
    public ShoppingBasket findByUserId(UserID userId) {
        return null;
    }

    @Override
    public void addItemToBasket(UserID userId, ProductID productID, int quantity) {

    }
}
